timemachine/fe/rbfe.py

```diff
@@ -125,34 +85,70 @@ def __init__(self, omm_system, conf, box):
         self.box = box
 
 
-def generate_samples(st, host_config, temperature, lambda_schedule, n_frames, seed):
+def sample(initial_state, protocol):
+    """
+    Generate a trajectory given an initial state and a simulation protocol
     """
-    Generate samples given a lambda_schedule. Let L be the number
-    of lambda windows T be the number of frames.
 
-    Parameters
-    ----------
-    st: SingleTopology
-        Topology container for the alchemical ligand
+    bound_impls = [p.bound_impl(np.float32) for p in initial_state.U_fns]
+    intg_impl = initial_state.integrator.impl()
+    baro_impl = initial_state.barostat.impl(bound_impls)
 
-    host_config: HostConfig
-        Configuration of the host
+    ctxt = custom_ops.Context(initial_state.x0, initial_state.v0, initial_state.box0, intg_impl, bound_impls, baro_impl)
 
-    temperature: float
-        Temperature of the system, in Kelvins
+    # burn-in
+    ctxt.multiple_steps_U(
+        lamb=initial_state.lamb,
+        n_steps=protocol.burn_in,
+        lambda_windows=[initial_state.lamb],
```

Comment:
No need to compute energies for burn in.

Reply:
i'm going to leave this in for debugging purposes -  sometimes its useful to monitor the progress of the nrgs

Reply:
Should we print these values then?

Reply:
I'm only printing these out when a simulation crashes - we probably don't want to print them out generally.

Reply:
also - I think `store_u_interval=0` should avoid computing the energies directly

Reply:
`store_u_interval=0` will run the energy on the last step. Not a big cost, just doesn't seem like it will be ever seen if run in production.